prov/util/src/util_shm.c

```diff
@@ -68,21 +68,36 @@ static void smr_peer_addr_init(struct smr_addr *peer)
 void smr_cma_check(struct smr_region *smr, struct smr_region *peer_smr)
 {
 	struct iovec local_iov, remote_iov;
+	int remote_pid;
 	int ret;
-
-	if (peer_smr->cma_cap != SMR_CMA_CAP_NA) {
-		smr->cma_cap = peer_smr->cma_cap;
-		return;
+	uint8_t mask, shift;
+
+	if (smr == peer_smr) {
+		mask = SMR_CMA_SELF_MASK;
+		shift = SMR_CMA_SELF_SHIFT;
+	} else {
+		mask = SMR_CMA_PEER_MASK;
+		shift = SMR_CMA_PEER_SHIFT;
+		if (peer_smr->cma_cap & SMR_CMA_SELF_MASK) {
```

Comment:
Shouldn't this be `SMR_CMA_PEER_MASK`?

Reply:
No. SMR_CMA_SELF_MASK (`#define SMR_CMA_SELF_MASK (0xF0)`) is used here to clear up low 4 bits (self_cma cap) and extract high 4 bits (peer_cma cap).